tests/gordo_components/builder/test_builder.py

```diff
@@ -456,7 +469,7 @@ def test_provide_saved_model_caching_handle_existing_different_register(tmp_dir)
     "should_be_equal,metadata,tag_list,replace_cache",
     [
         (True, None, None, False),
-        (False, {"metadata": "something"}, None, False),
+        (True, {"metadata": "something"}, None, False),
```

Comment:
wait, what happened here? Is the ` {"metadata": "something"}` available in `second_metadata` ? Because it should be

Reply:
Because we no longer count metadata as part of the cache key, right? Plus the addition of `build-metadata`as a subkey into `metadata` would break this.

Reply:
Closing #703 is nice, but then I propose you also add a test (or change this) to verify that even when the model is cached the second loading of the model gets the user-specified metadata attached as expected.